httpclient/httpclient.go

```diff
@@ -26,3 +29,30 @@ func CloneHTTPHeader(in http.Header) http.Header {
 	}
 	return out
 }
+
+// MustHTTPClient wraps delegate transports with logging, metrics, rate limiting, retryable, user agent, request id
+// and panics if any error occurs.
+func MustHTTPClient(cfg *Config, userAgent, reqType string, delegate http.RoundTripper) *http.Client {
+	if delegate == nil {
+		delegate = http.DefaultTransport
+	}
+
+	delegate = NewLoggingRoundTripper(delegate, reqType)
```

Comment:
I’m not sure it’s a good idea to log all requests unconditionally without providing a way to configure it.

Could we consider adding a configuration like the following?
```yaml
log:
  slowRequestThreshold: 1s # Log requests taking 1s or longer (warn level).
  mode: none | failed_requests | all_requests # Log modes:
                                 # - none: Disable logging entirely.
                                 # - failed_requests: Log only failed requests (4xx/5xx) at error level.
                                 # - all_requests: Log all requests (2xx at info level, 4xx/5xx at error level).
                                 # Default: failed_requests or none (to be decided).
```
@andrewesteves, what’s your opinion on this approach?